serveradmin/powerdns/templates/powerdns/domains.html

```diff
@@ -1,10 +1,64 @@
 {% extends "base.html" %}
 
 {% block content %}
-    <div class="container">
-        <h3>Settings</h3>
-        Configure domain and record settings to map Serveradmin objects to PowerDNS.
+    <div class="row">
+        <div class="col-md-2"></div>
+        <div class="col-md-8">
+            <h3>Mapping</h3>
+            <div class="alert alert-secondary" role="alert">
+                The following servertypes are mapped to PowerDNS domains.
+            </div>
 
-        <h2>Domain</h2>
+            {% for setting in settings %}
+            <div style="float: left; margin-right: 10px;">
+                <table class="table table-sm table-borderless table-striped table-responsive">
+                    <tbody>
+                        <tr>
+                            <td><b>{{ setting.servertype }}</b></td>
+                            <td></td>
+                        </tr>
+                        {% for name, value in setting.items %}
+                            <tr>
+                                <td>{{ name }}</td>
+                                <td>{{ value }}</td>
+                            </tr>
+                        {% endfor %}
+                    </tbody>
+                </table>
+            </div>
+            {% endfor %}
+        </div>
     </div>
-{% endblock %}
\ No newline at end of file
+    <div class="row">
+        <div class="col-md-2"></div>
+        <div class="col-md-8">
+            <h3>Domains</h3>
+            {% if out_of_sync|length %}
```

Comment:
`out_of_sync` is of type adminapi.dataset.Query which is lazy loaded. While I am not sure, I was hoping that using the length templatetag could prevent fetching the results and instead use a better way to count the objects internally.

Alltough if it does not (yet). Wouldn't it make sense to use it to clarify my intend and use the benefit if it ever gets added ?

Reply:
I see .. I wonder if passing a lazy loaded list into a template is maybe a bad practice .. but nothing to discuss now.

In general using `if list` or `if not list` is more pythonic than `if len(list)` or `if not len(list)` and that what I was aiming for here.

If you are really just interested in the length, then the view might should just pass the amount of out of sync domains to the template in the first place.

If you want to change that later on again, then your way might be the best way right now to avoid queryset hydration.